### PR TITLE
Nginx gzip configuration

### PR DESCRIPTION
**Pull request by JayH5**
Nginx provides the ability to gzip responses. This is a commonly used option but is not used in the current setup.

### CODE REVIEW

File: nginx/nginx.conf [hunk 1]
```diff
@@ -29,7 +29,48 @@ http {
 
     keepalive_timeout  65;
 
-    #gzip  on;
+    gzip              on;
+    gzip_disable      "msie6";
+    # gzip requests proxied from the load balancer
+    gzip_proxied      any;
```
Comment (JayH5) [inline nginx/nginx.conf]: It's more standard to _not_ gzip proxied requests as you generally don't want to gzip things that will be cached upstream. In our case we don't have any caching set up upstream yet so I'm just gzip-ing everything.

**Comment by bearnard** [comment]
We certainly should have this, my only concern is old phone browsers, in particular, those opera user-agent ones that crop up.

The internet reckons anything made in the last 15 years should be fine.
http://schroepl.net/projekte/mod_gzip/browser.htm mentions opera

**Comment by JayH5** [comment]
@bearnard I think we should be ok meeting those requirements 😅 .

My other argument would be that before our Molo sites used this image, they used Nginx with the default Debian config that already had gzip enabled. So the majority of sites that will be affected by this were already using gzip in the past for some time.

**Comment by bearnard** [comment]
👍 We should probably keep tabs on GA or something for a bit after.

**Comment by JayH5** [comment]
It'll take some time before this trickles down to running sites that build on this image, so I'm going to merge this for now and try check on the next site that updates to the latest Molo.

**Pull request merged**